pkg/webhooks/pod_webhook.go

```diff
@@ -113,16 +113,51 @@ func (p *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
 		} else {
 			groupUniqueKey = pod.Labels[leaderworkerset.GroupUniqueHashLabelKey]
 		}
-		_, foundEpKey := pod.Annotations[leaderworkerset.ExclusiveKeyAnnotationKey]
-		if foundEpKey && !exclusiveAffinityApplied(*pod) {
-			SetExclusiveAffinities(pod, groupUniqueKey)
+		epKey, foundEpKey := pod.Annotations[leaderworkerset.ExclusiveKeyAnnotationKey]
+		if (foundEpKey) && !exclusiveAffinityApplied(*pod, epKey) {
+			SetExclusiveAffinities(pod, groupUniqueKey, epKey, leaderworkerset.GroupUniqueHashLabelKey)
+		}
+		_, foundSubGroupSize := pod.Annotations[leaderworkerset.SubGroupSizeAnnotationKey]
+		if foundSubGroupSize {
```

Comment:
```suggestion
		if foundSubGroupSize && pod.Labels[leaderworkerset.SubGroupIndexLabelKey] != "" {
```
We should skip if the subgroup labels/annotation are already added